helix-core/src/syntax.rs

```diff
@@ -316,6 +317,14 @@ impl Syntax {
         syntax
     }
 
+    pub fn get_query(&self, query_filename: &str) -> Query {
+        let query = read_query(
+            &self.language_id.to_ascii_lowercase(),
+            &format!("{}.scm", query_filename),
+        );
+        Query::new(self.config.language, &query).unwrap()
+    }
+
```

Comment:
I'd rather see you revert the changes to Syntax and follow the indents query approach: https://github.com/helix-editor/helix/blob/3426285a6341702ace35817d38cefd2bb8b16437/helix-core/src/syntax.rs#L174-L182

Initializing and parsing a query is expensive so it should be memoized and only done once per language.

Reply:
Done 👍🏾